src/Abac.php

```diff
@@ -4,33 +4,60 @@
 
 use PhpAbac\Manager\{
     AttributeManager,
+    AttributeManagerInterface,
     CacheManager,
+    CacheManagerInterface,
     ComparisonManager,
-    PolicyRuleManager
+    ComparisonManagerInterface,
+    PolicyRuleManager,
+    PolicyRuleManagerInterface
 };
+use PhpAbac\Model\PolicyRule;
 use PhpAbac\Model\PolicyRuleAttribute;
 
 final class Abac
 {
-    /** @var PolicyRuleManager **/
+    /**
+     * @var PolicyRuleManager|PolicyRuleManager
+     */
     private $policyRuleManager;
-    /** @var AttributeManager **/
+    /**
+     * @var AttributeManager|AttributeManagerInterface
```

Comment:
It is not useful to set the Manager in PHPdoc if the interface is present. You can remove it :)

Reply:
hm.. i think you are not right. Phpdoc comments used by IDE such as PhpStorm  to show relations between different code. All logic of code-inspectors based on phpdocumentor comments. So if you will run inspector it will show a lot of errors "unknown classes" etc. 
Anyway relations of code is important thing if you decided to rename (refactor) some of class.
(Or in case when you call some class not directly, dynamically)

Reply:
Sorry I wasn't precise enough. I just said that you could do:

``/** @var AttributeManagerInterface **/`` and not mention `AttributeManager`` since the purpose to set interfaces instead of classes directly is to remove the dependency to a precise class.

I have no doubts on the utility of PHPdoc comments to declare properties type. It's just for methods where I suggest that PHP7 type-hinting is enough for modern IDEs

Reply:
oh, yes. But in this case IDE will not hint methods of some sibling class of interface. 
I mean quick jump to class method from call by ide shortcut ( ctrl+click ). If i describe parameter only as interface class ide jumps into interface class, not child. 
Anyway it's your code, your rules :-)

Reply:
There is no need to specify all siblings in DocBlock. Only interfaces is fine.

Reply:
The only one note. It's good practice to have FQCN (Fully Qualified Class Names) in DocBlocks.

Reply:
@abolabo The matter in open-source is not whose code it is, but what is the best way to code this library, and I'm totally open to changes even the ones I wouldn't do myself if it is the best thing according to conventions etc...

Your PR is good, it is totally right to apply SOLID principles here. You did transform the class type-hint into interfaces to remove hard dependencies between classes and it is essential to allow overriding of the library components. To be coherent with this change, the PHPdoc must also be updated, but must not keep the old hard dependency to the classes such as ``AttributeManager``.

In the meantime, you can remove the classes use statements which would become useless with the replacement by interfaces.

Reply:
I understand your concern about jumps in IDE but if someone overrides AttributeManager and set it as Abac dependency, it would be wrong that the IDE jumps to the native AttributeManager instead of the extended one. Jumping to the interface seems to be more coherent